modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ComponentAdapter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app;
+
+import org.apache.ignite.internal.close.ManuallyCloseable;
+import org.apache.ignite.internal.manager.IgniteComponent;
+
+/**
+ * Adapter for {@link ManuallyCloseable} to {@link IgniteComponent}.
+ */
+public class ComponentAdapter<T extends ManuallyCloseable> implements IgniteComponent {
```

Comment:
This adapter really doesn't look like a good architectural choice. IgniteComponent and ManuallyCloseable should not be converted into one another. I understand why you need this, but I would recommend to close the generator explicitly, it's not that hard